dask/dataframe/io/parquet/core.py

```diff
@@ -0,0 +1,452 @@
+
+    Notes
+    -----
+    Each partition will be written to a separate file.
+
+    Parameters
+    ----------
+    df : dask.dataframe.DataFrame
+    path : string
+        Destination directory for data.  Prepend with protocol like ``s3://``
+        or ``hdfs://`` for remote data.
+    engine : {'auto', 'fastparquet', 'pyarrow'}, default 'auto'
+        Parquet library to use. If only one library is installed, it will use
+        that one; if both, it will use 'fastparquet'.
+    compression : string or dict, optional
+        Either a string like ``"snappy"`` or a dictionary mapping column names
+        to compressors like ``{"name": "gzip", "values": "snappy"}``. The
+        default is ``"default"``, which uses the default compression for
+        whichever engine is selected.
+    write_index : boolean, optional
+        Whether or not to write the index. Defaults to True *if* divisions are
+        known.
+    append : bool, optional
+        If False (default), construct data-set from scratch. If True, add new
+        row-group(s) to an existing data-set. In the latter case, the data-set
+        must exist, and the schema must match the input data.
+    partition_on : list, optional
+        Construct directory-based partitioning by splitting on these fields'
+        values. Each dask partition will result in one or more datafiles,
+        there will be no global groupby.
+    storage_options : dict, optional
+        Key/value pairs to be passed on to the file-system backend, if any.
+    compute : bool, optional
+        If True (default) then the result is computed immediately. If False
+        then a ``dask.delayed`` object is returned for future computation.
+    **kwargs
+        Extra options to be passed on to the specific backend.
+
+    Examples
+    --------
+    >>> df = dd.read_csv(...)  # doctest: +SKIP
+    >>> to_parquet('/path/to/output/', df, compression='snappy')  # doctest: +SKIP
+
+    See Also
+    --------
+    read_parquet: Read parquet data to dask.dataframe
+    """
+    partition_on = partition_on or []
+    if isinstance(partition_on, string_types):
+        partition_on = [partition_on]
+
+    if set(partition_on) - set(df.columns):
+        raise ValueError("Partitioning on non-existent column")
+
+    if compression != "default":
+        kwargs["compression"] = compression
+    elif "snappy" in compress:
+        kwargs["compression"] = "snappy"
+
+    if isinstance(engine, str):
+        engine = get_engine(engine)
+
+    fs, fs_token, _ = get_fs_token_paths(
+        path, mode="wb", storage_options=storage_options
+    )
+    # Trim any protocol information from the path before forwarding
+    path = infer_storage_options(path)["path"]
+
+    if write_index:
+        df = df.reset_index()
+
+    out = engine.write(
+        df, fs, fs_token, path, append=append, partition_on=partition_on, **kwargs
+    )
+
+    if compute:
+        out.compute()
+        return None
```

Comment:
Has been suggested elsewhere that it might be useful to return the set of paths created.

Reply:
Sure!  Fixing.